daemon/daemon.go

```diff
@@ -781,7 +781,7 @@ func NewDaemon(ctx context.Context, config *config.Config, pluginStore *plugin.S
 	// initialization
 	defer func() {
 		if err != nil {
-			if err := d.Shutdown(ctx); err != nil {
+			if err := d.Shutdown(context.Background()); err != nil {
```

Comment:
This looks like a fix-up for the first commit, so would make sense to squash with that one.

Would it be useful to add a comment here, to describe we deliberately use a `context.Background()` here? Or do you think that would be clear from how it's used?

Reply:
Done, added a comment too. Thanks!